test-pad/test-pad

```diff
@@ -162,61 +163,68 @@ get_possible_node_roles() {
     esac
 }
 
-# RKE2 needs a delay between node startup
-# else the service crashes from too many learners
-rke2_delay() {
+# Both K3s and RKE2 need a needs a delay between node startup
+# else the service crashes from too many etcd learners
+node_delay() {
     if [ "${REPO}" = "rke2" ] && [ "$1" != "$2" ]; then
         sleep 30
+    elif [ "${REPO}" = "k3s" ] && [ "$1" != "$2" ]; then
+        sleep 10
     fi
 }
 
-while getopts 'r:c:v:b:i:p:kdohs' c; do
-    case $c in
-    r) 
-        REPO=$OPTARG
-    ;;
-    c) 
-        CLUSTER=$OPTARG
-    ;;
-    v) 
-        VERSION=$OPTARG
-    ;;
-    b)
-        BINARY_LOCATION=$OPTARG
-    ;;
-    i)
-        IMAGE_LOCATION=$OPTARG
-    ;;
-    k)
-        KILL=true
-    ;;
-    d)
-        DESTROY=true
-    ;;
-    s)
-        SKIP_DOWNLOAD=true
-    ;;
-    o)
-        ONLY_DOWNLOAD=true
-    ;;
-    p)
-        OS=$OPTARG
-    ;;
-    h)
-        usage
-        exit 0
-    ;;
-    *)
-        usage
+has_getopt() {
+    if [ -z "$(command -v getopt)" ]; then
+        printf "error: getopt is not installed\nif on MacOS: brew install gnu-getopt\n"
         exit 1
-    ;;
+    fi
+}
+
+if [ "$#" -eq 0 ]; then
+    usage
+    exit 0
+fi
+
+PARSED_ARGUMENTS=$(getopt -o r:v:c:b:i:kdh --long os:,version:,skip,download,help -- "$@")
+
+eval set -- "$PARSED_ARGUMENTS"
+
+while true; do
+    case "$1" in
+    -r)
+        REPO=$2; shift 2;;
+    -c)
+        CLUSTER=$2; shift 2;;
+    -v|--version)
+        VERSION=$2; shift 2;;
+    -b)
+        BINARY_LOCATION=$2; shift 2;;
+    -i) 
+        IMAGE_LOCATION=$2; shift 2;;
+    -k|--kill)
+        KILL=true; shift 1;;
+    -d)
+        DESTROY=true; shift 1;;
+    -s|--skip)
+        SKIP_DOWNLOAD=true; shift 1;;
+    --download)
+        ONLY_DOWNLOAD=true; shift 1;;
+    --os)
+        OSFLAG=$2; shift 2;;
+    -h|--help)
+        usage; exit 0;;
+    --) shift; break ;;
+    *) echo "Unexpected flag: $1"
+        usage; exit 1;;
     esac
 done
 
 has_vagrant
 has_curl
 has_wget
 has_awk
+has_getopt
```

Comment:
Does this need to appear above PARSED_ARGUMENTS?

Reply:
Good catch, I have move the `has_XXX` up to occur first